Gems/Atom/RHI/Vulkan/Code/Source/RHI/RenderPass.cpp

```diff
@@ -194,40 +196,42 @@ namespace AZ
                 template<AttachmentType type>
                 void BuildAttachmentReferences(uint32_t subpassIndex, SubpassInfo& subpassInfo) const
                 {
-                    AZStd::span<const RenderPass::SubpassAttachment> subpassAttachmentList = GetSubpassAttachments(subpassIndex, type);
-                    AZStd::vector<VkAttachmentReferenceType>& attachmentReferenceList =
-                        subpassInfo.m_attachmentReferences[static_cast<uint32_t>(type)];
-                    attachmentReferenceList.resize(subpassAttachmentList.size());
-                    for (uint32_t index = 0; index < subpassAttachmentList.size(); ++index)
+                    // A template cannot be specialized inside of another template
+                    // therefore if constexpr is used to filter the Preserve AttachmentType functionality
```

Comment:
```suggestion
                    // therefore constexpr is used to filter the Preserve AttachmentType functionality
```

Reply:
The "if" was not a typo.
The comment is meant to say "if constexpr" as in the `if constexpr` block in the code.